lib/pbench/test/unit/server/test_endpoint_configure.py

```diff
@@ -0,0 +1,42 @@
+from pbench.server.api.resources.query_apis import get_index_prefix
+
+
+class TestEnpointConfig:
+    """
+    Unit testing for EndpointConfig class.
+
+    In a web service context, we access class functions mostly via the
+    Flask test client rather than trying to directly invoke the class
+    constructor and `post` service.
+    """
+
+    def test_query(self, client, server_config):
+        """
+        test_query Check that endpoint data matches the config file.
+        """
+        host = server_config.get("pbench-server", "proxy_host")
+        uri_prefix = server_config.rest_uri
+        uri = f"{host}{uri_prefix}"
+        prefix = get_index_prefix(server_config)
+        expected_results = {
+            "metadata": {
+                "identification": f"Pbench server {server_config.COMMIT_ID}",
+                "prefix": prefix,
+                "run_index": f"{prefix}.v6.run-data.",
```

Comment:
Not sure this is correct for the dashboard today, as it is currently concatenating the prefix and the specific index where needed.  In the future, when we drop direct Elasticsearch queries from the dashboard, we'll be able to drop the need for this index names.  But for now, we can avoid additional code changes in the dashboard until then by not adding the prefix in the index names.

Reply:
The dashboard today doesn't use this; I have another branch where I started modifying the dashboard code to load the endpoints from this API, and of course it will directly reference the appropriate properties from the loaded JSON instead of constructing them as it does today.